dymension/libs/kaspa/lib/validator/src/deposit.rs

```diff
@@ -179,7 +181,21 @@ pub async fn validate_new_deposit_inner(
         });
     }
 
-    let containing_block: RpcBlock = client_node
+    // Connect to Kaspa node via gRPC to fetch the block
+    let grpc_url = kaspa_urls_grpc
+        .first()
+        .ok_or_else(|| ValidationError::KaspaNodeError {
+            reason: "No gRPC URL provided".to_string(),
+        })?
+        .clone();
+
+    let grpc_client = GrpcClient::connect(grpc_url)
+        .await
+        .map_err(|e| ValidationError::KaspaNodeError {
+            reason: format!("Failed to connect to gRPC: {}", e),
+        })?;
```

Comment:
you can't just make a new grpc conn on every request I don't think it's going to work well...